### ISSUE TITLE
Issue with KSP picking up Kotlin Compiler classpath dependencies

### ISSUE THREAD

**Issue by EdbertChan** [bazelbuild/rules_kotlin#1138]
We have a project where Kotlin and a Motif (KSP plugin) is being mixed in the same module. 

We are finding that Dagger will run and pick up the Guava classes that are packed in with the KotlinCompiler.jar that is supplied by Jetbrains.

I am working on a solid repro case but I am opening this issue in case anyone else has seen it.

Example:

Consider the below stack trace. I can see that when going through the hierarchy of the object being tested, they are loaded from places that are not supplied by the codebase or by Dagger. It looks like rules_kotlin is hijacking the classpath.

jar:file:/Users/edbert/.bazelcache/workspace/ff080e9ef5254374b7ce1c9f144b976d/execroot/__main__/bazel-out/xplat-opt-exec-ST-4ec9c722f6b9/bin/external/io_bazel_rules_kotlin/src/main/kotlin/build.runfiles/io_bazel_rules_kotlin/src/main/kotlin/jdeps-gen.jar!/com/google/common/graph/AbstractNetwork$1.class

jar:file:/Users/edbert/.bazelcache/workspace/ff080e9ef5254374b7ce1c9f144b976d/execroot/__main__/bazel-out/xplat-opt-exec-ST-4ec9c722f6b9/bin/external/io_bazel_rules_kotlin/src/main/kotlin/build.runfiles/io_bazel_rules_kotlin/src/main/kotlin/jdeps-gen.jar!/com/google/common/graph/AbstractBaseGraph.class

jar:file:/Users/edbert/.bazelcache/workspace/ff080e9ef5254374b7ce1c9f144b976d/execroot/__main__/bazel-out/xplat-opt-exec-ST-4ec9c722f6b9/bin/external/io_bazel_rules_kotlin/src/main/kotlin/build.runfiles/com_github_jetbrains_kotlin/lib/kotlin-compiler.jar!/com/google/common/graph/BaseGraph.class

Stack trace

```
error: [ksp] java.lang.ClassCastException: class com.google.common.graph.AbstractNetwork$1 cannot be cast to class com.google.common.graph.SuccessorsFunction (com.google.common.graph.AbstractNetwork$1 and com.google.common.graph.SuccessorsFunction are in unnamed module of loader org.jetbrains.kotlin.preloading.MemoryBasedClassLoader @370736d9)
        at com.google.common.graph.Traverser.<init>(Traverser.java:72)
        at com.google.common.graph.Traverser.<init>(Traverser.java:68)
        at com.google.common.graph.Traverser$1.<init>(Traverser.java:100)
        at com.google.common.graph.Traverser.forGraph(Traverser.java:100)
        at com.google.common.graph.Graphs.reachableNodes(Graphs.java:199)
        at dagger.internal.codegen.extension.DaggerGraphs.unreachableNodes(DaggerGraphs.java:94)
        at dagger.internal.codegen.binding.BindingGraphConverter.convert(BindingGraphConverter.java:81)
        at dagger.internal.codegen.binding.BindingGraphFactory.create(BindingGraphFactory.java:111)
        at dagger.internal.codegen.processingstep.ComponentProcessingStep.processRootComponent(ComponentProcessingStep.java:117)
        at dagger.internal.codegen.processingstep.ComponentProcessingStep.process(ComponentProcessingStep.java:88)
        at dagger.internal.codegen.processingstep.ComponentProcessingStep.process(ComponentProcessingStep.java:50)
        at dagger.internal.codegen.processingstep.TypeCheckingProcessingStep.lambda$process$0(TypeCheckingProcessingStep.java:82)
        at com.google.common.collect.RegularImmutableMap.forEach(RegularImmutableMap.java:185)
        at dagger.internal.codegen.processingstep.TypeCheckingProcessingStep.process(TypeCheckingProcessingStep.java:70)
        at dagger.internal.codegen.processingstep.TypeCheckingProcessingStep.process(TypeCheckingProcessingStep.java:48)
        at dagger.spi.shaded.androidx.room.compiler.processing.XProcessingStep.process(XProcessingStep.kt:59)
        at dagger.spi.shaded.androidx.room.compiler.processing.CommonProcessorDelegate.processRound(XBasicAnnotationProcessor.kt:130)
        at dagger.spi.shaded.androidx.room.compiler.processing.ksp.KspBasicAnnotationProcessor.process(KspBasicAnnotationProcessor.kt:62)
        at com.google.devtools.ksp.AbstractKotlinSymbolProcessingExtension$doAnalysis$8$1.invoke(KotlinSymbolProcessingExtension.kt:306)
        at com.google.devtools.ksp.AbstractKotlinSymbolProcessingExtension$doAnalysis$8$1.invoke(KotlinSymbolProcessingExtension.kt:304)
        at com.google.devtools.ksp.AbstractKotlinSymbolProcessingExtension.handleException(KotlinSymbolProcessingExtension.kt:410)
        at com.google.devtools.ksp.AbstractKotlinSymbolProcessingExtension.doAnalysis(KotlinSymbolProcessingExtension.kt:304)
        at org.jetbrains.kotlin.cli.jvm.compiler.TopDownAnalyzerFacadeForJVM.analyzeFilesWithJavaIntegration(TopDownAnalyzerFacadeForJVM.kt:112)
        at org.jetbrains.kotlin.cli.jvm.compiler.TopDownAnalyzerFacadeForJVM.analyzeFilesWithJavaIntegration$default(TopDownAnalyzerFacadeForJVM.kt:77)
        at org.jetbrains.kotlin.cli.jvm.compiler.KotlinToJVMBytecodeCompiler$analyze$1.invoke(KotlinToJVMBytecodeCompiler.kt:256)
        at org.jetbrains.kotlin.cli.jvm.compiler.KotlinToJVMBytecodeCompiler$analyze$1.invoke(KotlinToJVMBytecodeCompiler.kt:247)
        at org.jetbrains.kotlin.cli.common.messages.AnalyzerWithCompilerReport.analyzeAndReport(AnalyzerWithCompilerReport.kt:115)
        at org.jetbrains.kotlin.cli.jvm.compiler.KotlinToJVMBytecodeCompiler.analyze(KotlinToJVMBytecodeCompiler.kt:247)
        at org.jetbrains.kotlin.cli.jvm.compiler.KotlinToJVMBytecodeCompiler.compileModules$cli(KotlinToJVMBytecodeCompiler.kt:87)
        at org.jetbrains.kotlin.cli.jvm.compiler.KotlinToJVMBytecodeCompiler.compileModules$cli$default(KotlinToJVMBytecodeCompiler.kt:43)
        at org.jetbrains.kotlin.cli.jvm.K2JVMCompiler.doExecute(K2JVMCompiler.kt:165)
        at org.jetbrains.kotlin.cli.jvm.K2JVMCompiler.doExecute(K2JVMCompiler.kt:50)
        at org.jetbrains.kotlin.cli.common.CLICompiler.execImpl(CLICompiler.kt:104)
        at org.jetbrains.kotlin.cli.common.CLICompiler.execImpl(CLICompiler.kt:48)
        at org.jetbrains.kotlin.cli.common.CLITool.exec(CLITool.kt:101)
        at io.bazel.kotlin.compiler.BazelK2JVMCompiler.exec(BazelK2JVMCompiler.kt:32)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:566)
        at io.bazel.kotlin.builder.toolchain.KotlinToolchain$KotlinCliToolInvoker.compile(KotlinToolchain.kt:213)
        at io.bazel.kotlin.builder.tasks.jvm.CompilationTaskKt$runKspPlugin$1$1$1.invoke(CompilationTask.kt:303)
        at io.bazel.kotlin.builder.tasks.jvm.CompilationTaskKt$runKspPlugin$1$1$1.invoke(CompilationTask.kt:303)
        at io.bazel.kotlin.builder.toolchain.CompilationTaskContext.executeCompilerTask(CompilationTaskContext.kt:126)
        at io.bazel.kotlin.builder.toolchain.CompilationTaskContext.executeCompilerTask$default(CompilationTaskContext.kt:118)
        at io.bazel.kotlin.builder.tasks.jvm.CompilationTaskKt$runKspPlugin$1.invoke(CompilationTask.kt:301)
        at io.bazel.kotlin.builder.tasks.jvm.CompilationTaskKt$runKspPlugin$1.invoke(CompilationTask.kt:291)
        at io.bazel.kotlin.builder.toolchain.CompilationTaskContext.execute(CompilationTaskContext.kt:153)
        at io.bazel.kotlin.builder.toolchain.CompilationTaskContext.execute(CompilationTaskContext.kt:145)
        at io.bazel.kotlin.builder.tasks.jvm.CompilationTaskKt.runKspPlugin(CompilationTask.kt:291)
        at io.bazel.kotlin.builder.tasks.jvm.CompilationTaskKt.runPlugins(CompilationTask.kt:246)
        at io.bazel.kotlin.builder.tasks.jvm.KotlinJvmTaskExecutor.execute(KotlinJvmTaskExecutor.kt:52)
        at io.bazel.kotlin.builder.tasks.KotlinBuilder.executeJvmTask(KotlinBuilder.kt:216)
        at io.bazel.kotlin.builder.tasks.KotlinBuilder.build(KotlinBuilder.kt:107)
        at io.bazel.kotlin.builder.tasks.CompileKotlin.invoke(CompileKotlin.kt:27)
        at io.bazel.worker.PersistentWorker$workTo$1.invoke(PersistentWorker.kt:97)
        at io.bazel.worker.PersistentWorker$workTo$1.invoke(PersistentWorker.kt:97)
        at io.bazel.worker.WorkerContext$TaskContext.resultOf(WorkerContext.kt:128)
        at io.bazel.worker.WorkerContext.doTask(WorkerContext.kt:156)
        at io.bazel.worker.PersistentWorker$start$1$1$producer$1$1.invoke$lambda$0(PersistentWorker.kt:70)
        at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
        at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
        at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
        at java.base/java.lang.Thread.run(Thread.java:829)

```

**Comment by restingbull** [comment]
Should be resolved with #1167

**Comment by EdbertChan** [comment]
It fixes the original error but now I'm getting:

`exception: java.lang.IllegalAccessError: failed to access class com.google.common.graph.BaseGraph from class com.google.common.graph.Traverser (com.google.common.graph.BaseGraph is in unnamed module of loader org.jetbrains.kotlin.preloading.MemoryBasedClassLoader @370736d9; com.google.common.graph.Traverser is in unnamed module of loader java.net.URLClassLoader @f15525c)`
